src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "mouse.h"
+#include "mouse_core.h"
+
+#include <algorithm>
+
+#include "bios.h"
+#include "bitops.h"
+#include "callback.h"
+#include "checks.h"
+#include "cpu.h"
+#include "dos_inc.h"
+#include "int10.h"
+#include "pic.h"
+#include "regs.h"
+
+using namespace bit::literals;
+
+CHECK_NARROWING();
+
+// This file implements the DOS mouse driver interface,
+// using host system events
+
+// Reference:
+// - Ralf Brown's Interrupt List
+// - WHEELAPI.TXT from CuteMouse package
+// - https://www.stanislavs.org/helppc/int_33.html
+// - http://www2.ift.ulaval.ca/~marchand/ift17583/dosints.pdf
+
+#define CURSOR_SIZE_X  16
+#define CURSOR_SIZE_Y  16
+#define CURSOR_SIZE_XY (CURSOR_SIZE_X * CURSOR_SIZE_Y)
+
+#define NUM_BUTTONS 3
+
+enum class MouseCursor : uint8_t { Software = 0, Hardware = 1, Text = 2 };
+
+// These values represent 'hardware' state, not driver state
+
+MouseButtons12S buttons = 0;
+float pos_x     = 0.0f;
+float pos_y     = 0.0f;
+int16_t wheel   = 0;
+uint8_t rate_hz = 0; // TODO: add proper reaction for 0 (disable driver)
+
+static struct { // DOS driver state
+
+    // Structure containing (only!) data which should be
+    // saved/restored during task switching
+
+    // DANGER, WILL ROBINSON!
+    ///
+    // This whole structure can be read or written from the guest side
+    // via virtual DOS driver, functions 0x15 / 0x16 / 0x17.
+    // Do not put here any array indices, pointers, or anything that
+    // can crash the emulator if filled-in incorrectly, or that can
+    // be used by malicious code to escape from emulation!
+
+    bool enabled    = false; // TODO: make use of this
+    bool cute_mouse = false;
+
+    uint16_t times_pressed[NUM_BUTTONS]   = {0};
+    uint16_t times_released[NUM_BUTTONS]  = {0};
+    uint16_t last_released_x[NUM_BUTTONS] = {0};
+    uint16_t last_released_y[NUM_BUTTONS] = {0};
+    uint16_t last_pressed_x[NUM_BUTTONS]  = {0};
+    uint16_t last_pressed_y[NUM_BUTTONS]  = {0};
+    uint16_t last_wheel_moved_x           = 0;
+    uint16_t last_wheel_moved_y           = 0;
+
+    float mickey_x = 0.0f;
+    float mickey_y = 0.0f;
+
+    float mickeys_per_px_x = 0.0f;
+    float mickeys_per_px_y = 0.0f;
+    float pxs_per_mickey_x = 0.0f;
+    float pxs_per_mickey_y = 0.0f;
```

Comment:
`pxs` stands for pixels? Then just rename it to `pixels_per_mickey_x`. Saving a few keystrokes like this is rarely worth it as it makes it more difficult for other to decipher the intent behind things. In general, we should by all means optimise for long-term readability; "saving" a few keystrokes here and there is meaningless in that context.